packages/steveo/src/consumers/sqs.ts

```diff
@@ -235,6 +235,9 @@ class SqsRunner extends BaseRunner implements IRunner {
     await bluebird.map(
       subscriptions,
       async topic => {
+        // If the consumer is supposed to terminate
+        // do not process any more messages
+        if (this.manager.shouldTerminate) return;
```

Comment:
Any in-flight promise should also be stopped from further processing. 

Since we drain the pool now, it couldn't attain a resource but will error out, hence safe to just stop here.